portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,69 @@
 
 package com.google.sps.servlets;
 
+import com.google.gson.Gson;
+
+import java.util.ArrayList;
+import java.util.List;
+
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+    private List<String> messages;
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    messages = new ArrayList<>();
+    Query query = new Query("comment").addSort("time", SortDirection.DESCENDING);;
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+        for (Entity entity : results.asIterable()) {
+            String comment = (String)entity.getProperty("text");
+            messages.add(comment);
+        }
+    response.setContentType("application/json;");
+    String json = convertToJsonUsingGson(messages);
+    response.getWriter().println(json);
+  }
+
+    @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      String newComment = getComment(request);
+      long timestamp = System.currentTimeMillis();
+
+      Entity commentEntity = new Entity("comment");
+      commentEntity.setProperty("text",newComment);
+      commentEntity.setProperty("time", timestamp);
+
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      datastore.put(commentEntity);
+
+      response.sendRedirect("/index.html");
+  }
+
+  private String getComment(HttpServletRequest request) {
```

Comment:
This util function is quite simple, is used only once, and the function name getComment is a little bit confusing (e.g. just by looking at the function name, people may think it performs the functionality currently performed by doGet). What about removing this function and merge its functionality into doPost?